models/tests.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"github.com/go-xorm/core"
+	"github.com/go-xorm/xorm"
+	_ "github.com/mattn/go-sqlite3" // for the test engine
+)
+
+var engineCreated = false
+
+// CreateTestEngine create an xorm engine for testing
+func CreateTestEngine() error {
+	if engineCreated {
+		return nil
+	}
+	var err error
+	x, err = xorm.NewEngine("sqlite3", "file::memory:?cache=shared")
+	if err != nil {
+		return err
+	}
+	x.SetMapper(core.GonicMapper{})
+	if err = x.StoreEngine("InnoDB").Sync2(tables...); err != nil {
+		return err
+	}
+	engineCreated = true
+	return nil
+}
+
+func deleteTestBeans(deleteBeans... interface{}) error {
```

Comment:
Should we use [testfixtures](https://github.com/go-testfixtures/testfixtures) instead of manually deleting/inserting beans?

The biggest problem of doing that manually is having a lot of foreign key violations while running tests.